R/generate_retrospective_data.R

```diff
@@ -0,0 +1,99 @@
+#' Generate retrospective reporting triangles
+#'
+#' This function ingests a reporting triangle and the number of retrospective
+#'   reporting triangles we want to create, and iteratively generates the
+#'   reporting triangle that would have been available as of the maximum
+#'   reference time, and iteratively working backwards for
+#'   `n_history_uncertainty` snapshots
+#'
+#' @param triangle_for_retro Matrix of the incomplete reporting triangle
+#'   to be used to generate retrospective nowcasts, with rows representing the
+#'   time points of reference and columns representing the delays
+#' @param n_history_uncertainty Integer indicating the number of retrospective
```

Comment:
I don't really understand the uncertainty naming here? What is it referring to? the downstream use case?

Reply:
This is what filters to the number of datasets you need to estimate the uncertainty, it sounds like your preference is to pass back a less filtered version and remove/move all of these checks for having sufficient data to a downstream function

Reply:
As described above, this specification will be punted to the downstream function `generate_retro_nowcasts()`